File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
         HashMap<Integer, Integer> hm = new HashMap<>();
        
        for (int num : nums) {
            hm.put(num, hm.getOrDefault(num, 0) + 1);
        }
        
        for (int key : hm.keySet()) {
            if (hm.get(key) > nums.length / 2) {
                return key;
            }
        }
        return -1; 
    
    }
}

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int can=nums[0];
        int n=nums.size(); int count=1;
        for(int i=1;i<n;i++){
            if(nums[i]==can) count++;
            else count--;
            if(count==0){
                count=1;
                can=nums[i];
            }
        }
        return can;
    }
};